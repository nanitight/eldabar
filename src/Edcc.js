import React from  'react' ;
import { HashRouter,Route , Link, Switch} from 'react-router-dom' ;
import About from './components/About';
import Contact from './components/Contact';
import AddVideo from './AddVideo' ;
import './EDCC.css' ;
import './css/global.css' ;

import Playlist from './Playlist';
class EDCCNavBar extends React.Component{
    constructor(props){
        super(props) ;
        this.url = 'http://localhost:9009' ;
        // this.url= 'https://nd1server1.herokuapp.com'
        // this.state = {
        //     videos:[]
        // }
    }
    
    render(){
        return(
            <HashRouter>
                <div className="nav-container" data-testid="nav">
                    <Link className="click-btn title" to="/add"> 
                        <img id="logo" src={`${this.url}/artist/edcc/logosquare.jpg`} alt="logo"/>
                    </Link>

                    <div className="nav-list">
                        <Link to="/">HOME </Link>
                        <Link to="/about">About Us </Link>
                        <Link to="/contact">Contact Us </Link>
                    </div>
                </div>
                <div className="content" data-testid="content">
                <Switch>
                    <Route path="/" exact >
                        <Playlist url={this.url}/>
                    </Route>
                    <Route path="/contact" component={Contact}/>
                    <Route path="/about" component={About} />
                    <Route path="/add" component={AddVideo}/>
                </Switch>
                </div>
                <div id="footer" data-testid="footer">
                    <p id="slogan">
                        Empowering People To Impact The World
                    </p>
                        Copyright 2019-2021
                </div>
            </HashRouter>
        )
    }
}

export default EDCCNavBar ;