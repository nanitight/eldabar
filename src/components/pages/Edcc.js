import React from  'react' ;
import { withRouter,Route , Switch, Link} from 'react-router-dom' ;
import ImageSlider from '../inc/ImageSlider';
import img from '../images/scripture.jpg' ;
import logo from '../images/WLogo.jpeg'

import '../../css/global.css' ;

import Playlist from './Playlist';
import FacebookVideoList from './FacebookVideoList';
class EDCC extends React.Component{
    constructor(props){
        super(props) ;
        // this.url = 'http://localhost:9009' ;
        this.url= 'https://nd1server1.herokuapp.com'
        // this.state = {
        //     videos:[]
        // }
    }
    
    render(){
        const {path, url} = this.props.match ;
        console.log('EDCC props',this.props) ;
        return(<>
            <div className='bg-secondary bg-gradient bg-opacity-10'>
                <ImageSlider />
                <Switch>
                <Route path={`${path}/fvideos`} >
                   <FacebookVideoList />
                        
                </Route>    
                    <Route path={`${path}/videos`} >

                    <Playlist url={this.url}/>
                    </Route>
                    <Route exact component={() => {
                        return <>
                            <section className='py-4'>
                                <div className='container'>
                                    <div className='row'>
                                        <div className='col-md-12'>
                                            <h4>Our Church</h4>
                                            <div className="underline mb-3" ></div>
                                            <p>
                                                We believe that Priests are called to stand as intercessors between God and the
                                                communities they serve in the market place while kings are called to give the Kingdom
                                                vision and mission to those communities.<br />
                                                So the work of the priest is to bless the king to succeed in the market place battle,
                                                and the work of the king is to take the spoil in the market place and bring them in the
                                                Kingdom of God.
                                                In the Kingdom of God all are full time, whether as a pastor or medical doctor.
                                                WE ARE A FAMILY OF KINGS AND PRIESTS ADVANCING THE ADVANCEMENT OF THE KINGDOM OF GOD
                                            </p>
                                        </div>
                                    </div>


                                </div>
                            </section>

                            <section>
                                <div className='container '>
                                    <div className='row'>
                                        <div className='col-md-6'>
                                            <div className="card shadow" >
                                                <img src={img} className="card-img-top mx-auto py-3" alt="..." style={{ maxWidth: '250px' }} />
                                                <div className="card-body">
                                                    <h5 className="card-title">What To Expect From Our Service</h5>
                                                    <hr className='underline-sub' />
                                                    <div className="card-text">
                                                        <ul>
                                                            <li>
                                                                Teaching Session - congregants, get to ask questions during the session.
                                                            </li>
                                                            <li>
                                                                Worship and Praise - congregants sing along in praise and worship to our king Yeshua.
                                                            </li>
                                                            <li>
                                                                Preaching Session - congregants sit for the Word of God and encouraged.
                                                            </li>
                                                            <li>
                                                                Public Reading - Every member of the congregation reads the word of God out loud with others.
                                                            </li>
                                                            <li>
                                                                Offerings are accepted
                                                            </li>
                                                        </ul>
                                                        This is not neccesarily the strict order, led by the Holy Spirit, the program may change.
                                                    </div>
                                                    <Link className="btn btn-primary" to={`${url}/fvideos`}>Facebook Video</Link>
                                                </div>
                                            </div>
                                        </div>

                                        <div className='col-md-6'>
                                            <img src={logo} alt='logo' style={{ maxWidth: '60%' }} />
                                        </div>
                                    </div>

                                </div>
                            </section>
                    </>
                }} >
                    
                    </Route>
                </Switch>
                </div>
               
            

                {/* <Link className="click-btn title" to={`${path}/add`}> 
                        <img className='ratio ratio1x1' src={`${this.url}/artist/edcc/logosquare.jpg`} alt="logo"/>
                    </Link> */}
            </>
        )
    }
}

export default withRouter(EDCC) ;