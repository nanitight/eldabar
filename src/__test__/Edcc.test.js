import React from 'react' ;
import ReactDOM from 'react-dom' ;

import Edcc from '../Edcc' ;
import {render, cleanup} from '@testing-library/react' ;
import "@testing-library/jest-dom/extend-expect" ;
import renderer from 'react-test-renderer' ;

afterEach(cleanup) ; 


//Make sure it renders

it("Renders without crashing",()=>{
    //test
    const div = document.createElement("div") ; 
    ReactDOM.render(<Edcc/>,div) ;
}) ;

it("Renders contents correctly",()=>{
    const {getByTestId} = render(<Edcc />)
    expect(getByTestId('footer')).toHaveTextContent('Copyright') ; 
    expect(getByTestId('nav')).toHaveTextContent(/Home /i) ;
    expect(getByTestId('content')).not.toBeEmptyDOMElement() ;
}) ;


//Snapshot testing
it("matches snapshot",()=>{
   const tree = renderer.create(<Edcc />).toJSON() ; //convert to vertual dom objet

   expect(tree).toMatchSnapshot() ;
})