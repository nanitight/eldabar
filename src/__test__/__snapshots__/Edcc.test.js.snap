// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="nav-container"
    data-testid="nav"
  >
    <img
      alt="logo"
      id="logo"
      src="https://nd1server1.herokuapp.com/artist/edcc/logosquare.jpg"
    />
    <div
      className="nav-list"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        HOME 
      </a>
      <a
        href="/about"
        onClick={[Function]}
      >
        About Us 
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact Us 
      </a>
    </div>
    <a
      className="click-btn title"
      href="/add"
      onClick={[Function]}
    >
      + 
    </a>
  </div>,
  <div
    className="content"
    data-testid="content"
  >
    <div
      data-testid="playlist-content"
      id="bigContainer"
    >
      <div
        id="VideoPlayer"
      >
        <p
          className="title"
        >
          EL-Dabar : The Word Of God
        </p>
        <iframe
          allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
          allowFullScreen={true}
          className="vidContainer"
          src=""
          title="EL-Dabar : The Word Of God"
        />
        <div>
          <button
            className="click-btn"
            onClick={[Function]}
          >
            Previous Video
          </button>
          <button
            className="click-btn"
            onClick={[Function]}
          >
            Next Video
          </button>
        </div>
      </div>
      <div
        className="Playlist Manager"
        data-testid="playlist-manager"
      >
        <p
          className="title"
        >
          <span
            className="click-btn"
          >
            â˜°
          </span>
          EL-DABAR : THE WORD OF GOD
        </p>
        <div>
          <p>
            No Videos Selected
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div
    data-testid="footer"
    id="footer"
  >
    <p
      id="slogan"
    >
      Empowering People To Impact The World
    </p>
    Copyright 2019-2021
  </div>,
]
`;
